src/backend/distributed/commands/multi_copy.c

```diff
@@ -1751,7 +1752,24 @@ CitusCopyDestReceiverStartup(DestReceiver *dest, int operation,
 	LockShardListResources(shardIntervalList, ShareLock);
 
 	/* keep the table metadata to avoid looking it up for every tuple */
-	copyDest->tableMetadata = cacheEntry;
+	copyDest->partitionMethod = cacheEntry->partitionMethod;
+	copyDest->shardCount = cacheEntry->shardIntervalArrayLength;
+	copyDest->shardIntervalCache = (ShardInterval **)
+								   PointerArrayFromList(shardIntervalList);
```

Comment:
It might be worth adding a comment on why we're calling `PointerArrayFromList` instead of using `cacheEntry->sortedShardIntervalArray` to prevent a sleepy developer from optimising this step away in the future.